src/arrayMethodJoin.js

```diff
@@ -4,9 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (separator === null) {
+      for (let i = 0; i < this.length; i++) {
```

Comment:
There is a redundant code duplication inside both loops, which can be avoided by moving the condition related to the last element after the loop.